infra/aws/terraform/prow-build-cluster/iam.tf

```diff
@@ -15,7 +15,28 @@ limitations under the License.
 */
 
 module "iam" {
+  count = var.cluster_name == "prow-build-cluster" ? 1 : 0
+
   source = "./modules/iam"
 
   eks_admins = var.eks_admins
 }
+
+resource "aws_iam_role" "iam_cluster_viewer" {
+  name        = "TFProwClusterViewer"
+  description = "IAM role used to delegate access to prow-build-cluster"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "AWS" : data.aws_iam_user.eks_viewers[*].arn
+        },
+        "Action" : "sts:AssumeRole",
+        "Condition" : {}
+      }
+    ]
+  })
+}
```

Comment:
Can we create this resource in the IAM module?

Reply:
I would prefer to keep it here. It doesn't contain any AWS policy attached. Also, we should create similar roles for writer and admin.